### ISSUE TITLE
[BUG] Flashing failed

### ISSUE THREAD

**Issue by FrancYescO** [Renaud11232/OctoPrint-Marlin-Flasher#73]
-Upgraded to latest version
-Upgraded arduino-cli to 0.17.0 (i was using 0.6.0)
-Uploaded a sketch that was working on the old versione
-"Flashing failed" message appear after few seconds after "Compiling"

how to get more debug infos?

<img width="1004" alt="image" src="https://user-images.githubusercontent.com/4574593/114781753-62419180-9d79-11eb-9409-ee047fcd2f7a.png">

**Comment by Renaud11232** [comment]
Hello,

In OctoPrint settings, under the Logging section, you need to set the logging level to "DEBUG" for the plugin and then save :
![image](https://user-images.githubusercontent.com/13933802/114922938-8b6f2a00-9e2c-11eb-98f3-ef950cdcb25c.png)

Once this is done, you can redo the same operation, so it appears in the logs and link them in this issue.
Can you also send the zip file you uploaded ?

Thanks

**Comment by FrancYescO** [comment]
nvm, it was just failing compiling due to a missing library, i just miss out the old popup where it show more detailed information about the failing lines ;)

**Comment by Renaud11232** [comment]
I'm reopening since it should show the popup.

Can you send the zip file so I can see what happened ?

**Comment by FrancYescO** [comment]
[Marlin.zip](https://github.com/Renaud11232/OctoPrint-Marlin-Flasher/files/6321061/Marlin.zip)
here you go

**Comment by Renaud11232** [comment]
I think I found the issue, it seems to be an issue with `arduino-cli`.
Even when the compilation fails, arduino-cli does return an empty error output.
```json
{
  "compiler_out": "",
  "compiler_err": "",
  "builder_result": {
    "build_path": "C:\\Users\\gaspa\\AppData\\Local\\Temp\\arduino-sketch-1AF59F1FD8A66CA6FD1C3D0031EDAE86"
  },
  "success": false
}
```
I opened an issue a while ago for a similar problem,
I'll try to adapt my plugin until it's fixed in arduino-cli

**Comment by FrancYescO** [comment]
For me to get some debug info I just stole with ps the command that marlin flasher was issuing, and after issuing the command on the terminal I noticed the compile error output (that was echoed before the JSON dictionary)

**Comment by fritzw** [comment]
I'd suggest to add more helpful output to the popup in case the compilation or flashing fails. For example, I got the following popup which contains no information at all. Not even a hint, how to proceed  to get more relevant information.
![marlin-flasher-error](https://user-images.githubusercontent.com/3628933/119636208-b332b400-be14-11eb-944b-847e5ef25360.png)

When enabling the debug log as explained above, I got the following, which contains the actual problem (PlatformIO outdated). 

Maybe it could help to show at least the first 100 lines of stdout in the error message popup when an error occurs (limited to a fixed value in order to prevent excessive memory usage during compilation)?

```
[...]
2021-05-25 16:42:25,421 - octoprint.plugins.marlin_flasher - INFO - Compiling...
2021-05-25 16:42:25,423 - octoprint.plugins.marlin_flasher - DEBUG - Executing command : /home/pi/.platformio/penv/bin/platformio run -d /home/pi/.octoprint/data/marlin_flasher/firmware_platformio/Marlin
2021-05-25 16:42:28,864 - octoprint.plugins.marlin_flasher - DEBUG - Return code : 1
2021-05-25 16:42:28,865 - octoprint.plugins.marlin_flasher - DEBUG - Standard output : Processing melzi (platform: atmelavr; board: sanguino_atmega1284p; framework: arduino)
--------------------------------------------------------------------------------
Verbose mode can be enabled via `-v, --verbose` option

**************************************************
******      An update to PlatformIO is      ******
******  required to build Marlin Firmware.  ******
******                                      ******
******      Minimum version:  (5, 0, 3)     ******
******      Current Version:  (4, 3, 4)     ******
******                                      ******
******   Update PlatformIO and try again.   ******
**************************************************
2021-05-25 16:42:28,865 - octoprint.plugins.marlin_flasher - DEBUG - Error output : ========================== [FAILED] Took 1.85 seconds ==========================
Environment               Status    Duration
------------------------  --------  ------------
mega2560                  IGNORED
mega1280                  IGNORED
MightyBoard1280           IGNORED
[...]
```

**Comment by Renaud11232** [comment]
@fritzw I've opened a separate issue for that problem (#76) since this issue is related to an issue with arduino-cli
